lib/debug.py

```diff
@@ -0,0 +1,197 @@
+            except socket.error, error:
+                self.log("Debugging: connection to pydevd failed: %s", error)
+                self.log("Be sure to run 'Start debugging server' in PyDev to enable debugging.")
+            else:
+                self.log("Enabled PyDev debugger.")
+
+            sys.stderr = stderr
+
+    def init_svg(self):
+        self.svg = etree.Element("svg", nsmap=inkex.NSS)
+        atexit.register(self.save_svg)
+
+    def save_svg(self):
+        tree = etree.ElementTree(self.svg)
+        with open(os.path.join(os.path.dirname(os.path.dirname(__file__)), "debug.svg"), "w") as debug_svg:
+            tree.write(debug_svg)
+
+    @check_enabled
+    def add_layer(self, name="Debug"):
+        layer = etree.Element("g", {
+            INKSCAPE_GROUPMODE: "layer",
+            INKSCAPE_LABEL: name,
+            "style": "display: none"
+        })
+        self.svg.append(layer)
+        self.current_layer = layer
+
+    @check_enabled
+    def open_group(self, name="Group"):
+        group = etree.Element("g", {
+            INKSCAPE_LABEL: name
+        })
+
+        self.log_svg_element(group)
+        self.group_stack.append(group)
+
+    @check_enabled
+    def close_group(self):
+        if self.group_stack:
+            self.group_stack.pop()
+
+    @check_enabled
+    def log(self, message, *args):
+        if self.last_log_time:
+            message = "(+%s) %s" % (datetime.now() - self.last_log_time, message)
+
+        self.raw_log(message, *args)
+
+    def raw_log(self, message, *args):
+        now = datetime.now()
+        timestamp = now.isoformat()
+        self.last_log_time = now
+
+        print >> self.log_file, timestamp, message % args
+        self.log_file.flush()
+
+    def time(self, func):
+        def decorated(*args, **kwargs):
+            if self.enabled:
+                self.raw_log("entering %s()", func.func_name)
+                start = time.time()
+
+            result = func(*args, **kwargs)
+
+            if self.enabled:
+                end = time.time()
+                self.raw_log("leaving %s(), duration = %s", func.func_name, round(end - start, 6))
+
+            return result
+
+        return decorated
+
+    @check_enabled
+    def log_svg_element(self, element):
+        if self.current_layer is None:
+            self.add_layer()
+
+        if self.group_stack:
+            self.group_stack[-1].append(element)
+        else:
+            self.current_layer.append(element)
+
+    @check_enabled
+    def log_line_string(self, line_string, name=None, color=None):
+        """Add a Shapely LineString to the SVG log."""
+        self.log_line_strings([line_string], name, color)
+
+    @check_enabled
+    def log_line_strings(self, line_strings, name=None, color=None):
+        path = line_strings_to_path(line_strings)
+        path.set('style', formatStyle({"stroke": color or "#000000", "stroke-width": "0.3"}))
+
+        if name is not None:
```

Comment:
I think
```
if not name
```

is more pythonic

Reply:
I could do `if name`, but that would be false for an empty string.  I guess it arguably isn't all that useful to set an empty string as a label, but I don't see why it shouldn't be possible.

I generally seem to like to compare things with `is None` when I know the thing might be None just to make it clear what I'm doing.

All that is to say I'm on the fence on this one, but I think I'll leave it.